File: src/main/java/be4rjp/sclat/api/GlowingAPI.java
package be4rjp.sclat.api;

import be4rjp.sclat.Main;
import com.comphenix.protocol.PacketType;
import com.comphenix.protocol.events.PacketContainer;
import com.comphenix.protocol.wrappers.WrappedDataWatcher;
import org.bukkit.entity.Entity;
import org.bukkit.entity.Player;

public class GlowingAPI {
	public static void setGlowing(Entity entity, Player player, boolean flag) {
		PacketContainer packet = Main.protocolManager.createPacket(PacketType.Play.Server.ENTITY_METADATA);
		packet.getIntegers().write(0, entity.getEntityId());
		WrappedDataWatcher watcher = new WrappedDataWatcher();
		WrappedDataWatcher.Serializer serializer = WrappedDataWatcher.Registry.get(Byte.class);
		watcher.setEntity(entity);
		watcher.setObject(0, serializer, (byte) (flag ? 0x40 : 0));
		packet.getWatchableCollectionModifier().write(0, watcher.getWatchableObjects());
		try {
			Main.protocolManager.sendServerPacket(player, packet);
		} catch (RuntimeException e) {
			e.printStackTrace();
		}
	}
}

File: src/main/kotlin/be4rjp/sclat/api/GlowingAPI.kt
@file:Suppress("PLATFORM_CLASS_MAPPED_TO_KOTLIN")

package be4rjp.sclat.api

import be4rjp.sclat.api.packet.Packets
import com.comphenix.protocol.PacketType
import com.comphenix.protocol.wrappers.WrappedDataWatcher
import org.bukkit.entity.Entity
import org.bukkit.entity.Player
import java.lang.Byte
import java.lang.reflect.InvocationTargetException
import kotlin.Boolean
import kotlin.Suppress

object GlowingAPI {
    @JvmStatic
    fun setGlowing(
        entity: Entity,
        player: Player,
        flag: Boolean,
    ) {
        val packet = Packets.createPacket(PacketType.Play.Server.ENTITY_METADATA)
        packet.integers.write(0, entity.entityId)
        val watcher = WrappedDataWatcher()
        val serializer = WrappedDataWatcher.Registry.get(Byte::class.java)
        watcher.entity = entity
        watcher.setObject(0, serializer, (if (flag) 0x40 else 0).toByte())
        packet.watchableCollectionModifier.write(0, watcher.watchableObjects)
        try {
            Packets.sendServerPacket(player, packet)
        } catch (e: InvocationTargetException) {
            e.printStackTrace()
        }
    }
}
